### PR TITLE
Dynamic Dashboard: Clean up legacy dashboard flow

### LINKED ISSUE #12601: Clean up files from the legacy UI flow

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #12601 

## Description

Since we migrated the dashboard screen to the SwiftUI views, we need to clean up the legacy UIKit code. This PR removes all the outdated code so it's quite large. Most of the changes are deletion of unused files. There are also some small changes in `DashboardViewModel` and `StoreStatsPeriodViewModel` to remove unused methods.

## Testing instructions

Smoke test the dynamic dashboard features following #12563 to confirm that everything still works correctly.

## Screenshots

N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/201">18.5</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StatsTimeRangeV4+UI.swift [hunk 1]
```diff
@@ -79,25 +79,4 @@ extension StatsTimeRangeV4 {
...
-        let dateFormatter: DateFormatter
-        switch intervalGranularity {
-        case .hourly:
-            dateFormatter = DateFormatter.Charts.chartAxisHourFormatter
-        case .daily:
-            dateFormatter = DateFormatter.Charts.chartAxisDayFormatter
-        case .weekly:
-            dateFormatter = DateFormatter.Charts.chartAxisDayFormatter
-        case .monthly:
-            dateFormatter = DateFormatter.Charts.chartAxisMonthFormatter
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StatsTimeRangeV4+UI.swift:96]: nit: `chartAxisMonthFormatter` and `chartAxisHourFormatter` are not used anywhere else. Should we consider deleting those from `WooCommerce/Classes/Extensions/DateFormatter+Helpers.swift`?

File: WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift [hunk 2]
```diff
@@ -104,12 +98,6 @@ final class DashboardViewModel: ObservableObject {
         installPendingThemeIfNeeded()
     }
 
-    /// Uploads the answers from the store creation profiler flow
-    ///
-    func uploadProfilerAnswers() async {
-        await storeCreationProfilerUploadAnswersUseCase.uploadAnswers()
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift:110]: `storeCreationProfilerUploadAnswersUseCase` property can be deleted as well.

**Review by selanthiraiyan — APPROVED** [review]
This looks good to me. I left a couple of comments about deleting more code. 🚢

File: WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StatsTimeRangeV4+UI.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StatsTimeRangeV4+UI.swift:96]: Nice catch! Removed in 26a5371d46b308e2b1ee2a714be2dd2ebfe0479a.

File: WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift:110]: Removed in d125d80ef40a2ed8f3b2e1eff225b63446d28dd7.

**Pull request merged**